APMrover2/commands_logic.cpp

```diff
@@ -396,14 +403,67 @@ void Rover::nav_set_yaw_speed()
 void Rover::do_wait_delay(const AP_Mission::Mission_Command& cmd)
 {
     condition_start = millis();
-    condition_value  = cmd.content.delay.seconds * 1000;    // convert seconds to milliseconds
+    condition_value = cmd.content.delay.seconds * 1000;    // convert seconds to milliseconds
 }
 
 void Rover::do_within_distance(const AP_Mission::Mission_Command& cmd)
 {
     condition_value  = cmd.content.distance.meters;
 }
 
+void Rover::do_yaw(const AP_Mission::Mission_Command& cmd)
+{
+    // Only support target yaw for now
+    condition_start = condition_value;  // save condition_value from current navigation wp loaded
+    // get final angle, 1 = Relative, 0 = Absolute
+    if (cmd.content.yaw.relative_angle == 0) {
+        // absolute angle
+        condition_value = cmd.content.yaw.angle_deg * 100;
+    } else {
+        // relative angle
+        condition_value = cmd.content.yaw.angle_deg * 100;
+        if (cmd.content.yaw.direction < 0) {
+            condition_value = -condition_value;
+        }
+
+        condition_value = condition_value + ahrs.yaw_sensor;
+    }
+
+    // absolute angle error
+    const int32_t error_to_target_yaw = abs((condition_value - ahrs.yaw_sensor));
+
+    // Calculate the steering to apply base on error calculated
+    const int32_t steering = steerController.get_steering_out_angle_error(error_to_target_yaw);
+    SRV_Channels::set_output_scaled(SRV_Channel::k_steering, steering);
+    next_navigation_leg_cd = condition_value;
+    calc_throttle(g.speed_cruise);
+
+    do_auto_rotation = true;
+}
+
+bool Rover::do_yaw_rotation()
+{
```

Comment:
So this function stops forward progress of the rover until it has turned the appropriate amount at which point we resume normal navigation